library/pod-security-policy/selinux/template.yaml

```diff
@@ -56,89 +56,147 @@ spec:
                     description: "An SELinux user."
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspselinux
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: usingAllowedSELinuxOptions
+            expression: |
+              has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.seLinuxOptions) ? 
+                (has(variables.params.allowedSELinuxOptions) ? 
+                (
+                  size(variables.params.allowedSELinuxOptions) > 0 && variables.params.allowedSELinuxOptions.all(opts,
+                    (has(opts.level) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.level) && (variables.anyObject.spec.securityContext.seLinuxOptions.level == opts.level) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.level)) &&
+                    (has(opts.role) ? has(variables.anyObject.spec.securityContext.seLinuxOptions.role) && (variables.anyObject.spec.securityContext.seLinuxOptions.role == opts.role) : !has(variables.anyObject.spec.securityContext.seLinuxOptions.role)) &&
```

Comment:
As mentioned in previous comment, this seems incorrect. It looks like you are disabling the check if the params side is undefined, as opposed to skipping if securityContext side is undefined.

Previous comment:

```
This test looks wrong. Shouldn't we be assuming an option is disallowed unless specifically allowed by parameters?

As-written, it looks like undefined parameters => allow everything, which would seem to drift from the rego.

Adding gator tests for this would validate the drift.
```

Please add the gator tests for this scenario, relying on reviewers to catch this drift consistently is brittle.